### PR TITLE
fix(robustness): CI fingerprint dedup + paginated gh retry amplification (#1669, #1670)

### LINKED ISSUE #1669: robustness: Multiple unnamed CI check runs silently deduplicated, dispatching only one fix mission
## Problem

When `fail.get("name", "unknown")` returns `"unknown"` for multiple different failing check runs (e.g., GitHub omits the `name` field on certain event types, or returns `null`), `compute_ci_fingerprint(pr_number, head_sha, "unknown")` produces the same hash for all of them. The dedup check at line 284 (`if fp_key in tracker: continue`) then skips all but the first. The unique check run `id` — available at `fail["id"]` and already used at line 288 — is not included in the fingerprint key.

## Why This Matters

Legitimate CI failures are silently dropped: only the first failing check run triggers a fix mission. The remaining failures are never addressed until a new commit is pushed. This can stall PR merges indefinitely when CI fails with unnamed jobs (e.g., composite actions, matrix jobs with dynamic names).

## Suggested Fix

Include the check run ID in the fingerprint to guarantee uniqueness:
```python
def compute_ci_fingerprint(pr_number, head_sha, job_name, run_id=""):
    key = f"{pr_number}:{head_sha}:{job_name}:{run_id}"
    return hashlib.sha256(key.encode()).hexdigest()[:16]
```
Call it as `compute_ci_fingerprint(pr_number, head_sha, job_name, str(fail.get("id", "")))`.

## Details

| | |
|---|---|
| **Severity** | 🟡 Medium |
| **Category** | robustness |
| **Location** | `koan/app/ci_dispatch.py:280-282` |
| **Effort** | ⚡ Quick fix |

---
🤖 Created by Kōan from audit session

### PR DESCRIPTION
**Pull request by tomgreen981111-cipher**
## What
Two independent robustness fixes batched together.

## Why

**#1669** — Multiple unnamed CI check runs (job name falls back to `"unknown"`) all produced the same dedup fingerprint (`pr:sha:unknown`), so only the first unnamed failure triggered a fix mission. The rest were silently dropped until a new commit was pushed.

**#1670** — `fetch_unresolved_review_comments()` and `fetch_review_body_comments()` used `--paginate` with `timeout=15`. Combined with `retry_with_backoff` (3 retries × 15s), a single slow GitHub API call could block the notification loop for ~45s per call, per PR, per project.

## How

**#1669** — Added `run_id: str = ""` parameter to `compute_ci_fingerprint()`. The call site now passes `str(fail.get("id", ""))`, making each check run's fingerprint unique even when names collide.

**#1670** — Replaced `--paginate` with `--limit 100` (GitHub's max-per-page). One API page eliminates retry amplification with no practical data loss — PRs with 100+ review comments are essentially nonexistent.

## Testing
- Added `test_unnamed_jobs_differ_by_run_id` and `test_run_id_defaults_to_empty` to `TestComputeCiFingerprint`
- Updated `test_dedup_prevents_double_dispatch` to generate the correct fingerprint (now includes `run_id="1"`)
- All 56 tests in `test_ci_dispatch.py` + `test_review_comment_dispatch.py` pass
- Full test suite passes

Closes #1669, #1670

### CODE REVIEW

**Review by atoomic — APPROVED** [review]

**Comment by BabyKoan** [comment]
## PR Review — fix(robustness): CI fingerprint dedup + paginated gh retry amplification (#1669, #1670)

Solid robustness fixes. Both changes are minimal, targeted, and well-tested.

- #1669: Adding `run_id` to fingerprint correctly fixes dedup collision for unnamed CI jobs. Default parameter `run_id: str = ""` maintains backward compatibility.
- #1670: Replacing `--paginate` with `--limit 100` eliminates retry amplification. Tradeoff is acceptable given PRs rarely exceed 100 review comments.
- Tests cover new behavior: `test_unnamed_jobs_differ_by_run_id`, `test_run_id_defaults_to_empty`, and updated `test_dedup_prevents_double_dispatch`.
- No security issues, no edge cases missed, no performance regressions introduced.

---

---

### Checklist

- [x] No hardcoded secrets
- [x] Input validation at boundaries
- [x] Error handling and cleanup
- [x] No mutable default arguments
- [x] Test coverage for new code paths
- [x] No N+1 queries or unbounded collections


---
_Automated review by [Kōan](https://koan.anantys.com)_ _(Ollama-launch · model qwen3.5:cloud)_ `HEAD=c4601c4` `33s`

**Review by BabyKoan — APPROVED** [review]
No blocking issues found.

File: koan/app/review_comment_dispatch.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def fetch_unresolved_review_comments(
     try:
         raw = run_gh(
             "api", f"repos/{full_repo}/pulls/{pr_number}/comments",
-            "--paginate", "--jq",
+            "--limit", "100", "--jq",
```
Comment (atoomic) [inline koan/app/review_comment_dispatch.py:115]: this is using a different output, but LGTM as we were not using the pagination in any of these calls

**Review by atoomic — APPROVED** [review]
